ts/client/src/clientIxParamBuilder.ts

```diff
@@ -110,3 +111,173 @@ export const NullPerpEditParams: PerpEditParams = {
+  ixGate.ior(p.AltExtend ? new BN(0) : new BN(1).ushln(5));
+  ixGate.ior(p.AltSet ? new BN(0) : new BN(1).ushln(6));
+  ixGate.ior(p.FlashLoan ? new BN(0) : new BN(1).ushln(7));
+  ixGate.ior(p.GroupClose ? new BN(0) : new BN(1).ushln(8));
+  ixGate.ior(p.GroupCreate ? new BN(0) : new BN(1).ushln(9));
+
+  ixGate.ior(p.PerpCancelAllOrders ? new BN(0) : new BN(1).ushln(10));
+  ixGate.ior(p.PerpCancelAllOrdersBySide ? new BN(0) : new BN(1).ushln(11));
+  ixGate.ior(p.PerpCancelOrder ? new BN(0) : new BN(1).ushln(12));
+  ixGate.ior(
+    p.PerpCancelOrderByClientOrderId ? new BN(0) : new BN(1).ushln(13),
+  );
+  ixGate.ior(p.PerpCloseMarket ? new BN(0) : new BN(1).ushln(14));
+  ixGate.ior(p.PerpConsumeEvents ? new BN(0) : new BN(1).ushln(15));
+  ixGate.ior(p.PerpCreateMarket ? new BN(0) : new BN(1).ushln(16));
+  ixGate.ior(p.PerpDeactivatePosition ? new BN(0) : new BN(1).ushln(17));
+  ixGate.ior(p.PerpLiqBasePosition ? new BN(0) : new BN(1).ushln(18));
+  ixGate.ior(p.PerpLiqForceCancelOrders ? new BN(0) : new BN(1).ushln(19));
+
+  ixGate.ior(p.PerpLiqQuoteAndBankruptcy ? new BN(0) : new BN(1).ushln(20));
+  ixGate.ior(p.PerpPlaceOrder ? new BN(0) : new BN(1).ushln(21));
+  ixGate.ior(p.PerpSettleFees ? new BN(0) : new BN(1).ushln(22));
+  ixGate.ior(p.PerpSettlePnl ? new BN(0) : new BN(1).ushln(23));
+  ixGate.ior(p.PerpUpdateFunding ? new BN(0) : new BN(1).ushln(24));
+  ixGate.ior(p.Serum3CancelAllOrders ? new BN(0) : new BN(1).ushln(25));
+  ixGate.ior(p.Serum3CancelOrder ? new BN(0) : new BN(1).ushln(26));
+  ixGate.ior(p.Serum3CloseOpenOrders ? new BN(0) : new BN(1).ushln(27));
+  ixGate.ior(p.Serum3CreateOpenOrders ? new BN(0) : new BN(1).ushln(28));
+  ixGate.ior(p.Serum3DeregisterMarket ? new BN(0) : new BN(1).ushln(29));
+
+  ixGate.ior(p.Serum3EditMarket ? new BN(0) : new BN(1).ushln(30));
+  ixGate.ior(p.Serum3LiqForceCancelOrders ? new BN(0) : new BN(1).ushln(31));
+  ixGate.ior(p.Serum3PlaceOrder ? new BN(0) : new BN(1).ushln(32));
+  ixGate.ior(p.Serum3RegisterMarket ? new BN(0) : new BN(1).ushln(33));
+  ixGate.ior(p.Serum3SettleFunds ? new BN(0) : new BN(1).ushln(34));
+  ixGate.ior(p.StubOracleClose ? new BN(0) : new BN(1).ushln(35));
+  ixGate.ior(p.StubOracleCreate ? new BN(0) : new BN(1).ushln(36));
+  ixGate.ior(p.StubOracleSet ? new BN(0) : new BN(1).ushln(37));
+  ixGate.ior(p.TokenAddBank ? new BN(0) : new BN(1).ushln(38));
+  ixGate.ior(p.TokenDeposit ? new BN(0) : new BN(1).ushln(39));
+
+  ixGate.ior(p.TokenDeregister ? new BN(0) : new BN(1).ushln(40));
+  ixGate.ior(p.TokenLiqBankruptcy ? new BN(0) : new BN(1).ushln(41));
+  ixGate.ior(p.TokenLiqWithToken ? new BN(0) : new BN(1).ushln(42));
+  ixGate.ior(p.TokenRegister ? new BN(0) : new BN(1).ushln(43));
+  ixGate.ior(p.TokenRegisterTrustless ? new BN(0) : new BN(1).ushln(44));
+  ixGate.ior(p.TokenUpdateIndexAndRate ? new BN(0) : new BN(1).ushln(45));
+  ixGate.ior(p.TokenWithdraw ? new BN(0) : new BN(1).ushln(46));
```

Comment:
this was very annoying to verify, is there a way to remove the repetition? (optional)